tests/e2e/taqueria-plugin-taquito.spec.ts

```diff
@@ -0,0 +1,346 @@
+import {generateTestProject} from "./utils/utils";
+import fs from "fs";
+import {execSync} from "child_process";
+import path from "path";
+import waitForExpect from "wait-for-expect";
+import { TezosToolkit } from '@taquito/taquito';
+
+
+describe("E2E Testing for taqueria taquito plugin",  () => {
+
+    const tezos = new TezosToolkit('https://hangzhounet.api.tez.ie');
+    const taqueriaProjectPath = 'e2e/auto-test-taquito-plugin';
+    const contractRegex = new RegExp("^\\w{36}$");
+    let environment: string;
+
+    beforeAll(async () => {
+        await generateTestProject(taqueriaProjectPath, ["taquito"]);
+    })
+
+    test('Verify that taqueria taquito plugin can deploy one contract using deploy command', async () => {
+        try {
+            environment = "test";
+            let smartContractHash = "";
+
+            // 1. Copy config.json and michelson contract from data folder to artifacts folder under taqueria project
+            execSync(`cp e2e/data/config-taquito-test-environment.json ${taqueriaProjectPath}/.taq/config.json`);
+            execSync(`cp e2e/data/hello-tacos.tz ${taqueriaProjectPath}/artifacts/`);
+
+            // Sometimes running two deploy commands in a short period of time cause an issue
+            // More details about the error - "counter %x already used for contract %y"
+            // It is applicable only for auto-tests that run very fast
+            // Uses retry mechanism to avoid test to fail
+
+            await waitForExpect(() => {
+
+                // 2. Run taq deploy on a selected test network described in "test" environment
+                const [,,,stdoutDeploy] = execSync(`taq deploy -e ${environment}`, {cwd: `./${taqueriaProjectPath}`}).toString().trim().split(/\r?\n/);
+
+                // 3. Verify that contract has been originated on the network
+                expect(stdoutDeploy).toContain("hello-tacos.tz");
+                expect(stdoutDeploy).toContain("hangzhounet");
+                const [,,contractHash] = stdoutDeploy.split("│");
+                smartContractHash = contractHash.trim();
+                expect(contractHash.trim()).toMatch(contractRegex);
+            });
+
+            // 4. Verify that contract has been originated to the network
+            await waitForExpect(async () => {
+                const contract = await tezos.contract.at(smartContractHash);
+                expect(contract.address).toBe(smartContractHash);
+            });
+
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
+
+    });
```

Comment:
We should add an additional step here that verifies that no other contracts were originated. That is, because 'taq originate [contract]` and `taq origiante` are both valid, we're not testing whether the first is properly interpreted.

Reply:
We will do it on sandbox, not on the network - need to add additional tests there. On the network, we will check that the second contract does not exist by checking the table.

Reply:
There is an issue with taquito-flextesa plugins tests -https://github.com/ecadlabs/taqueria/issues/405